server/graphql/v1/mutations/collectives.js

```diff
@@ -177,6 +178,74 @@ export async function createCollective(_, args, req) {
   return collective;
 }
 
+export async function createCollectiveFromGithub(_, args, req) {
+  if (!req.remoteUser) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in to create a collective',
+    });
+  }
+
+  if (!args.collective.name) {
+    throw new errors.ValidationFailed({ message: 'collective.name required' });
+  }
+
```

Comment:
TODO: we should do some extra validation of the GitHub repo/organization.

- check 100 stars
- validate that the user is an admin of the repo/organization
- check if the repo/organization is already registered as a collective

Reply:
- Checking stars already happens through filtering the repositories returned by GitHub.

- Currently the best way to know if a user is an admin is by fetching individual repo information, Github does not provide that information in list of public repos API currently being used. I'm still thinking about the best way to do this. While minding network round trips. 

- Yes this is important because it is included in the design, I will work on that.

Reply:
Technically the "100 stars" is currently checked on the frontend, so we can't trust it. That, and checking if the user is an admin of the repo/organization should be done again in the backend.

Good news, we have similar behavior elsewhere that we can copy!

Here, we check that the user is an admin:
https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L406

Here, we check the stars:
https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/orders.js#L153

Reply:
> Here, we check that the user is an admin:

Works well for a single repository fetch, it has `{permission: {admin: true} }` included in the repository data returned, in our case we are fetching all public repositories the data returned has no `permission` object included to validate `isAdmin`, the question is how do we do this for public repositories's list?:
Suggestions I'm thinking:
- Fetch all repositories, filter based on their stars and for each filtered repository fetch their `permission` data individually. The only problem is the user will have to wait for each repo in the array, before the `GithubRepositories` component is rendered. Time here is directly proportional to numbers of filtered repositories.

- My second suggestion is fetching the permission data only when a user selects a repo in `GithubRepositories` component and notify them, that, they don't have permission to create collective with the currently selected repo.

Reply:
If we’re checking that in the GraphQL mutation, we can do a single repository fetch, like in the examples I shared.